TrackingProduction/Fun4All_FieldOnAllTrackersCalos.C

```diff
@@ -0,0 +1,411 @@
+/*
+ * This macro shows a minimum working example of running the tracking
+ * hit unpackers with some basic seeding algorithms to try to put together
+ * tracks, calorimeter cluster reconstruction, and track projected to claorimeter surface. There are some analysis modules run at the end which package
+ * hits, clusters, and clusters on tracks, calo clusters, and towers in calo clusters into trees for analysis.
+ * also provide the option to do full tracking detectors or TPC only, and drift velocity correction and EMcal radius correction
+ */
+
+#include <G4_ActsGeom.C>
+#include <G4_Magnet.C>
+#include <GlobalVariables.C>
+#include <Trkr_Clustering.C>
+#include <Trkr_RecoInit.C>
+#include <Trkr_Reco.C>
+#include <G4_Global.C>
+
+#include <caloreco/RawClusterBuilderTopo.h>
+
+#include <ffamodules/CDBInterface.h>
+#include <fun4all/Fun4AllDstInputManager.h>
+#include <fun4all/Fun4AllDstOutputManager.h>
+#include <fun4all/Fun4AllInputManager.h>
+#include <fun4all/Fun4AllUtils.h>
+#include <fun4all/Fun4AllOutputManager.h>
+#include <fun4all/Fun4AllRunNodeInputManager.h>
+#include <fun4all/Fun4AllServer.h>
+
+#include <phool/recoConsts.h>
+
+#include <track_to_calo/TrackCaloMatch.h>
+#include <track_to_calo/TrackToCalo.h>
+#include <track_to_calo/CaloOnly.h>
+#include <track_to_calo/TrackOnly.h>
+
+#include <trackreco/AzimuthalSeeder.h>
+#include <trackreco/PHActsTrackProjection.h>
+#include <trackingdiagnostics/TrackResiduals.h>
+
+#include <trackbase_historic/SvtxTrack.h>
+
+#include <stdio.h>
+#include <iostream>
+#include <filesystem>
+
+R__LOAD_LIBRARY(libfun4all.so)
+R__LOAD_LIBRARY(libffamodules.so)
+R__LOAD_LIBRARY(libmvtx.so)
+R__LOAD_LIBRARY(libintt.so)
+R__LOAD_LIBRARY(libtpc.so)
+R__LOAD_LIBRARY(libmicromegas.so)
+R__LOAD_LIBRARY(libTrackingDiagnostics.so)
+R__LOAD_LIBRARY(libtrack_reco.so)
+R__LOAD_LIBRARY(libtrack_to_calo.so)
+R__LOAD_LIBRARY(libcalo_reco.so)
+
+using namespace std;
+
+std::string GetFirstLine(std::string listname);
+
+void Fun4All_FieldOnAllTrackersCalos(
+    const int nEvents = 10, 
+    vector<string> myInputLists = {
+        "run46730_0000_trkr.txt",
+        "run46730_calo.list"}, 
+    bool doTpcOnlyTracking = true,
+    bool doDriftVelocityCorr = true,
+    bool doEMcalRadiusCorr = true,
+    const bool convertSeeds = false)
+{
+  int verbosity = 0;
+
+  gSystem->Load("libg4dst.so");
+  //gSystem->Load("libFROG.so");
+  //FROG *fr = new FROG();
+
+  std::string firstfile = GetFirstLine(myInputLists[0]);
+  if (*firstfile.c_str() == '\0') return;
+  std::pair<int, int> runseg = Fun4AllUtils::GetRunSegment(firstfile);
+  int runnumber = runseg.first;
+  int segment = runseg.second;
+
+  auto rc = recoConsts::instance();
+  rc->set_IntFlag("RUNNUMBER", runnumber);
+
+  Enable::CDB = true;
+  rc->set_StringFlag("CDB_GLOBALTAG", "ProdA_2024");
+  rc->set_uint64Flag("TIMESTAMP", 6);
```

Comment:
This line will break any cdb lookup. It will use the calibrations for run number 6 (which is a simulation run). Any runwise calibration for the e.g. calos will be wrong

Reply:
Here, I refer to Fun4All_FieldOnAllTrackers.C. Do you have any suggestions to modify?

Reply:
It needs to be updated there too. We should set it to the actual run number

Reply:
`  Enable::CDB = true;
  rc->set_StringFlag("CDB_GLOBALTAG", "ProdA_2024");
  rc->set_uint64Flag("TIMESTAMP", runnumber);`
  Is it okay?